sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -153,6 +152,17 @@ class V2SessionCatalog(sessionState: SessionState) extends TableCatalog {
     }
   }
 
+  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = {
+    if (Option(loadTable(newIdent)).isEmpty) throw new TableAlreadyExistsException(newIdent)
+
+    Option(loadTable(oldIdent)) match {
```

Comment:
when loadTable returns null? I read loadTable, seems it throws NoSuchTableException when table doesn't exist, no?

Reply:
I got thrown off by https://github.com/apache/spark/pull/25206/files#diff-662d75c51e01d8e70f7c1384cea5b4c3L141 that checks for null.